### PR TITLE
duckdb: enable extensions

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Bump duckdb from 0.3.1 to 0.3.2 and enable extensions.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cpcloud** [comment]
@veprbl Would you mind reviewing this?

File: pkgs/development/libraries/duckdb/default.nix [hunk 1]
```diff
@@ -1,26 +1,65 @@
...
+    "-DBUILD_FTS_EXTENSION=${enableFeature withFts}"
+    "-DBUILD_HTTPFS_EXTENSION=${enableFeature withHttpFs}"
+    "-DBUILD_ICU_EXTENSION=${enableFeature withIcu}"
+    "-DBUILD_ODBC_DRIVER=${enableFeature withOdbc}"
+    "-DBUILD_PARQUET_EXTENSION=${enableFeature withParquet}"
+    "-DBUILD_REST_EXTENSION=${enableFeature withRest}"
+    "-DBUILD_TPCDS_EXTENSION=${enableFeature withTpcDs}"
+    "-DBUILD_TPCH_EXTENSION=${enableFeature withTpcH}"
+    "-DBUILD_VISUALIZER_EXTENSION=${enableFeature withVisualizer}"
+    "-DJDBC_DRIVER=${enableFeature withJdbc}"
```
Comment (veprbl) [inline pkgs/development/libraries/duckdb/default.nix:50]: As usual, I suggest not introducing excessive options
```suggestion
    "-DBUILD_FTS_EXTENSION=ON"
    "-DBUILD_HTTPFS_EXTENSION=${enableFeature withHttpFs}"
    "-DBUILD_ICU_EXTENSION=ON"
    "-DBUILD_ODBC_DRIVER=${enableFeature withOdbc}"
    "-DBUILD_PARQUET_EXTENSION=ON"
    "-DBUILD_REST_EXTENSION=ON"
    "-DBUILD_TPCDS_EXTENSION=ON"
    "-DBUILD_TPCH_EXTENSION=ON"
    "-DBUILD_VISUALIZER_EXTENSION=ON"
    "-DJDBC_DRIVER=${enableFeature withJdbc}"
```

**Review by veprbl — APPROVED** [review]

File: pkgs/development/libraries/duckdb/default.nix (hunk 1 above)
Comment (cpcloud) [reply pkgs/development/libraries/duckdb/default.nix:50]: Yup, thanks.

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 159526` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>duckdb</li>
  </ul>
</details>

**Pull request merged**